production/helm/loki/templates/_helpers.tpl

```diff
@@ -943,8 +943,11 @@ enableServiceLinks: false
 {{- define "loki.querySchedulerAddress" -}}
 {{- $schedulerAddress := ""}}
 {{- $isDistributed := eq (include "loki.deployment.isDistributed" .) "true" -}}
+{{- $isSimpleScalable := eq (include "loki.deployment.isScalable" .) "true" -}}
 {{- if $isDistributed -}}
 {{- $schedulerAddress = printf "%s.%s.svc.%s:%s" (include "loki.querySchedulerFullname" .) .Release.Namespace .Values.global.clusterDomain (.Values.loki.server.grpc_listen_port | toString) -}}
+{{- else if $isSimpleScalable -}}
+{{- $schedulerAddress = printf "query-scheduler-discovery.%s.svc.%s.:%s" .Release.Namespace .Values.global.clusterDomain (.Values.loki.server.grpc_listen_port | toString) -}}
```

Comment:
shouldn't it use a `read` service instead?

Reply:
I believe it should be the address of the backend, since in SSD mode it is 
Additionally, before Loki `3.0.0` release it was set to `query-scheduler-discovery` - [ref](https://github.com/grafana/loki/blob/v2.9.7/production/helm/loki/templates/_helpers.tpl#L804-L806)

But there was some more clarifications in the [issue](https://github.com/grafana/loki/issues/12506#issuecomment-2050700539), I'll do a bit more testing today and see if that could be left out empty. I had errors in the reader and writer when upgrading, but will check if I can resolve it.

Reply:
oops, yes, the `backend` target :smile: